src/_common/fireside/post/post-model.ts

```diff
@@ -345,6 +393,24 @@ export class FiresidePost extends Model implements ContentContainerModel {
 		return this.$_save(`/web/posts/manage/publish/${this.id}`, 'firesidePost');
 	}
 
+	$pin(targetModel: string) {
+		const $payload = this.$_save(
+			`/web/posts/manage/toggle-pin/${this.id}/${targetModel}`,
+			'post'
+		);
+		this.is_pinned = true; // Set this manually, the request does not gather this field.
```

Comment:
you shouldn't have to do this. `$_save` assigns the result back to the model.

Reply:
In the backend endpoint it's pretty difficult to figure out the context, that's why this is.